tools/validation-generator/src/main/java/io/spine/tools/validate/field/ConstraintCode.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.validate.field;
+
+import com.squareup.javapoet.CodeBlock;
+import io.spine.code.proto.FieldDeclaration;
+import io.spine.logging.Logging;
+import io.spine.tools.validate.AccumulateViolations;
+import io.spine.tools.validate.FieldAccess;
+import io.spine.tools.validate.MessageAccess;
+import io.spine.tools.validate.code.BooleanExpression;
+import io.spine.tools.validate.code.IsSet;
+
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.tools.validate.FieldAccess.element;
+import static io.spine.tools.validate.code.Blocks.empty;
+import static java.lang.System.lineSeparator;
+
+/**
+ * A validation constraint based on a Protobuf option of a message field.
+ */
+public final class ConstraintCode implements Logging {
```

Comment:
It's also weird that something called `...Code` isn't in the `.code` package. That also leads to the over-exposure of `Blocks` which otherwise would be package-private.